web/html/src/manager/proxy/container-config.tsx

```diff
@@ -0,0 +1,377 @@
+                : Network.errorMessageByStatus(xhr.status)[0];
+            setSuccess(false);
+            setMessages([errMessages]);
+            setLoading(false);
+          }
+        }
+      );
+    });
+  };
+
+  const clearFields = () => {
+    setModel(initialModel);
+  };
+
+  const onValidate = (isValidated: boolean) => {
+    setIsValidated(isValidated);
+  };
+
+  const onChange = (newModel: any) => {
+    setModel(Object.assign({}, newModel));
+  };
+
+  const onAddField = (fieldName: string) => {
+    return (index: number) => setModel(Object.assign({}, model, { [fieldName + index]: "" }));
+  };
+
+  const onRemoveField = (fieldName: string) => {
+    return (index: number) => {
+      const newModel = { ...model };
+      delete newModel[`${fieldName}${index}`];
+      setModel(newModel);
+    };
+  };
+
+  return (
+    <TopPanel
+      title={t("Container Based Proxy Configuration")}
+      icon="fa fa-cogs"
+      helpUrl="reference/proxy/container-based-config.html"
+    >
+      <p>
+        {t(
+          "You can generate a set of configuration files and certificates in order to register and run a container-based proxy. Once the following form is filled out and submitted you will get a .zip archive to download."
+        )}
+      </p>
+      {ContainerConfigMessages(success, messages, loading)}
+      <Form
+        className="form-horizontal"
+        model={model}
+        onValidate={onValidate}
+        onChange={onChange}
+        onSubmit={onSubmit}
+        title={t("Container Based Proxy Configuration")}
+      >
+        <Text
+          name="proxyFQDN"
+          label={t("Proxy FQDN")}
+          required
+          placeholder={t("e.g., proxy.domain.com")}
+          labelClass="col-md-3"
+          divClass="col-md-6"
+        />
+        <Text
+          name="serverFQDN"
+          label={t("Server FQDN")}
+          required
+          placeholder={t("e.g., server.domain.com")}
+          hint={t("FQDN of the server of proxy to connect to.")}
+          labelClass="col-md-3"
+          divClass="col-md-6"
+        />
+        <Text
+          name="proxyPort"
+          label={t("Proxy SSH port")}
+          hint={t("Port range: 1 - 65535")}
+          validators={[Validation.isInt({ gt: 0, lt: 65536 })]}
+          defaultValue="8022"
+          labelClass="col-md-3"
+          divClass="col-md-6"
+          type="number"
+        />
+        <Text
+          name="maxSquidCacheSize"
+          label={t("Max Squid cache size (MB)")}
+          required
+          validators={[Validation.isInt({ gt: 0 })]}
+          placeholder={t("e.g., 2048")}
+          labelClass="col-md-3"
+          divClass="col-md-6"
+          type="number"
+        />
+        <Text
+          name="proxyAdminEmail"
+          label={t("Proxy administrator email")}
+          validators={[Validation.matches(/^[a-zA-Z0-9.!#$%&’*+/=?^_`{|}~-]+@[a-zA-Z0-9-]+(?:\.[a-zA-Z0-9-]+)*$/)]}
```

Comment:
I would advise against this validation regex. For one, it's impossible to debug and understand, and for two it almost definitely is not complete. [See here](https://stackoverflow.com/a/201378/1470607) for one valid example, but I'm not sure if we should adopt that magic regex in our codebase either. In my opinion, we should either not validate the address at all, or do it with a third-party dependency, but that is just an opinion, I'm open to other ideas.